File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        
        mismatch = []
        if s1 == s2:
            return True
        for i in range(len(s1)):
            if s1[i] != s2[i]:
                mismatch.append(s1[i])
                mismatch.append(s2[i])
        
        if len(mismatch) != 4:
            return False
        if mismatch[0] == mismatch[3] and mismatch[1] == mismatch[2]:
            return True
        else:
            return False

File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        char c1 = '\0';
        char c2 = '\0';
        char c3 = '\0';
        char c4 = '\0';
        int count = 0;
        if (s1 == s2) return true;
        for (int i = 0; i < s1.size(); i++){
            if (s1[i] != s2[i]){
                count++;
                if (c1 != '\0') {
                    c2 = s1[i];
                    c4 = s2[i];
                }else{
                    c1 = s1[i];
                    c3 = s2[i];
                }
            }
        }
        if (count != 2) return false;
        if (c1 == c4 && c2 == c3) return true;
        return false;
    }
};